### PR TITLE
WT-6583 Fix read timestamp being cleared under cache pressure

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/include/txn.i [hunk 1]
```diff
@@ -1048,26 +1049,35 @@ __wt_txn_begin(WT_SESSION_IMPL *session, const char *cfg[])
...
 
         __wt_txn_get_snapshot(session);
     }
 
-    F_SET(txn, WT_TXN_RUNNING);
     if (F_ISSET(S2C(session), WT_CONN_READONLY))
         F_SET(txn, WT_TXN_READONLY);
 
-    return (0);
+    if (0) {
```
Comment (agorrod) [inline src/include/txn.i:1076]: We primarily use this pattern when there is some error cleanup shared with the fail/non-fail case, and some that is specific to the failure case. A pattern like in cur_backup.c:353 would be more appropriate here.

Please also update the wiki coding standards to reflect this - it's not obvious, and I doubt it's made explicit there.

File: test/suite/test_rollback_to_stable10.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def simulate_crash_restart(self, olddir, newdir):
         self.session = self.setUpSessionOpen(self.conn)
 
     def test_rollback_to_stable(self):
-        nrows = 1000
+        nrows = 2000
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply test/suite/test_rollback_to_stable10.py:80]: This change is reliably causing the failure. I'll try to create another test to reproduce this.

File: src/include/txn.i [hunk 3]
```diff
@@ -1048,26 +1049,35 @@ __wt_txn_begin(WT_SESSION_IMPL *session, const char *cfg[])
     if (cfg != NULL)
         WT_RET(__wt_txn_config(session, cfg));
 
+    /*
+     * Mark the transaction as running before obtaining the snapshot to avoid the read timestamp
+     * being cleared in eviction under cache pressure.
+     */
+    F_SET(txn, WT_TXN_RUNNING);
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply src/include/txn.i:1056]: It's cleared in __wt_hs_cursor_close. We have logic in cursor reset to check how many cursors are held by the session. If it holds no cursor, we will try to release its snapshot.

```
    /*
     * When the count of active cursors in the session goes to zero, there are no active cursors,
     * and we can release any snapshot we're holding for read committed isolation.
     */
    if (session->ncursors == 0 && !F_ISSET(cbt, WT_CBT_NO_TXN))
        __wt_txn_read_last(session);
```

I think it is more risky to fix the problem here as I'm not familiar with this part of the subsystem.

File: src/include/txn.i (hunk 1 above)
Comment (quchenhao) [reply src/include/txn.i]: Fixed and updated coding style in wiki.

**Comment by quchenhao** [comment]
@agorrod Can you look at this?

**Comment by quchenhao** [comment]
Close in favour of #5963

**Pull request closed without merging**